### ISSUE TITLE
historyApiFallback not doing rewrites in webpack 5

### ISSUE THREAD

**Issue by superheri** [webpack/webpack-dev-server#2798]
- Operating System: Windows 10
- Node Version: 12.18.4
- NPM Version: 6.14.6
- webpack Version: 5.2.0
- webpack-dev-server Version: 3.11.0
- Browser: Google Chrome 86.0.4240.111

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

```js
// webpack.config.js

devServer : {
    index: 'index.html',
    historyApiFallback: {
        rewrites: [
            {from: /a\/.*/, to: 'a/index.html'},
            {from: /b\/.*/, to: 'b/index.html'}
        ]
    }
}
```

### Expected Behavior

I expect this to actually return me to `a/index.html` when I do whatever url starting with `a/`.
I expect the same thing with `b`.
I also expect the 'index' option to be the file returned in any other case.

### Actual Behavior

It always return me the base `index.html`... Whatever value in `index` is the one returned.
From what it seems, it is like the code is doing a
```js
if (historyApiFallback) {...}
```
somewhere and it always acts like we wrote `true` in the `historyApiFallback` setting.

In webpack 4, it all worked well. I can't use only one file in index.html in my case because I need 2 'apps' (each with a different path), to be running at the same time.

### For Bugs; How can we reproduce the behavior?
Just reproduce a webpack.config.js with some config like my exemple and you can reproduce the bug easy.

**Comment by evilebottnawi** [comment]
No changes between webpack@4 and webpack@5 for this option and I can't reproduce

**Comment by superheri** [comment]
I even tried to use the 'proxy' setting, but it does not work. Do you think it has a link with the use of `webpack serve` instead of `webpack-dev-server`? Should I stick with webpack-cli 3?

**Comment by evilebottnawi** [comment]
> Do you think it has a link with the use of webpack serve instead of webpack-dev-server?

I think no, how you run webpack-dev-server?

> Should I stick with webpack-cli 3?

No, you will have a lot of problems

Maybe something wrong with your environment

**Comment by superheri** [comment]
I run my webpack-dev-server like this in webpack 5 :
"watch": "webpack-cli serve --mode=development --config ./webpack/all.webpack.config.js",

I have nothing really special in my config: 

```js
output: {
            path: path.resolve(baseFolder, 'dist'),
            chunkFilename: '[name].[chunkhash].js'
        },
devServer: {
            index: 'index.html',
            contentBase: path.resolve(baseFolder, 'dist'),
            writeToDisk: true,
            port: 3000,
            https: false,
            compress: true,
            historyApiFallback: {
                rewrites: [
                    { from: /main\/.*/, to: 'main/index.html' },
                    { from: /portal\/.*/, to: 'portal/index.html' }
                ]
            },
            host: '0.0.0.0',
            disableHostCheck: true
        }
```

Maybe it has to do with the fact that my main index.html il copied by CopyWebpackPlugin?

**Comment by evilebottnawi** [comment]
@superheri can you try to switch on `webpack-cli@3` and test again, and try to switch on `webpack@4` and test again and provide test results?

**Comment by superheri** [comment]
Webpack 4.44.2 and Webpack-cli 3.3.12 works well, I was using this with the same configuration before trying to migrate to webpack 5

I just tried running with Webpack 5.2.0 and Webpack-cli 3.3.12 and it now works just like before. It really seems like Webpack cli is messing up with my options...

Should I send this to the webpack-cli repo?

Thank you for taking the time to answer me by the way 👍

**Comment by evilebottnawi** [comment]
Yes, look like it is bug on webpack-cli side, let's report it t webpack-cli repo and I will look at this, will be great if you provide minimum reproducible test repo, so I will add your case to tests and we never faces with it in future

**Comment by superheri** [comment]
Good, I'll try to create a test repo soon. I'll tell you when it is done

**Comment by superheri** [comment]
@evilebottnawi A test repo is now available here with a readme telling how to reproduce

https://github.com/superheri/webpack-cli-bug
